gem/gem.py

```diff
@@ -782,6 +783,79 @@ def __new__(cls, i, j):
         return self
 
 
+# TODO: inverse should be different from comp tensor
+# in the sense that it does not take a local to a global view of a matrix
+# but instead eats a global view and spits out another global view
+# it does not transformation between free indices and shape so you would think it does not need a multiindex
+# but the multiindex is necessary to feed in the global view of the tensor
+# it can still have free_indices inherited by the  aggregate (?)
+class Inverse(Node):
+    __slots__ = ('children', 'multiindex', 'shape')
+    __back__ = ('multiindex',)
+
+    def __new__(cls, aggregate, multiindex):
+        assert multiindex
+
+        # Set index extents from shape
+        for index, extent in zip(multiindex, aggregate.shape):
+            assert isinstance(index, Index)
+            index.set_extent(extent)
+
+        self = super(Inverse, cls).__new__(cls)
+        self.children = (aggregate,)
+        self.shape = aggregate.shape
+        self.multiindex = multiindex
+        self.free_indices = aggregate.free_indices
```

Comment:
So you inherit shape and inherit free indices (both are fine), but there is still this multiindex...

Reply:
Addressed in c2a473f24b50396278f27cd4ecd1a6d4bfef0219